src/cpu/dyn_cache.h

```diff
@@ -788,31 +788,31 @@ static void cache_init(bool enable) {
 				cache_blocks[i].cache.next = &cache_blocks[i + 1];
 			}
 		}
-		if (cache_code_start_ptr==NULL) {
+		if (cache_code_start_ptr==nullptr) {
 			// allocate the code cache memory
 #if defined (WIN32)
-			cache_code_start_ptr=(Bit8u*)VirtualAlloc(0,cache_code_size,
+			cache_code_start_ptr=static_cast<uint8_t*>(VirtualAlloc(0,cache_code_size),
 				MEM_COMMIT,PAGE_EXECUTE_READWRITE);
 			if (!cache_code_start_ptr)
-				cache_code_start_ptr=reinterpret_cast<Bit8u*>(malloc(cache_code_size));
+				cache_code_start_ptr=static_cast<uint8_t*>(malloc(cache_code_size));
 #else
 			int map_flags = MAP_PRIVATE | MAP_ANON;
 			int prot_flags = PROT_READ | PROT_WRITE | PROT_EXEC;
 #if defined(HAVE_MAPJIT)
 			map_flags |= MAP_JIT;
 #endif
 #if defined(HAVE_MMAP)
-			cache_code_start_ptr=(Bit8u*)mmap(nullptr, cache_code_size, prot_flags, map_flags, -1, 0);
+			cache_code_start_ptr=static_cast<uint8_t*>(mmap(nullptr, cache_code_size, prot_flags, map_flags, -1, 0));
 			if (cache_code_start_ptr == MAP_FAILED)
 				E_Exit("Allocating dynamic core cache memory failed with errno %d", errno);
 #else
-			cache_code_start_ptr=reinterpret_cast<Bit8u*>(malloc(cache_code_size));
+			cache_code_start_ptr=static_cast<uint8_t*>(malloc(cache_code_size));
 			if (!cache_code_start_ptr)
 				E_Exit("Allocating dynamic core cache memory failed");
 #endif
 #endif
 			// align the cache at a page boundary
-			cache_code = (Bit8u *)(((Bitu)cache_code_start_ptr +
+			cache_code = reinterpret_cast<uint8_t*>(((Bitu)cache_code_start_ptr +
 			                        PAGESIZE_TEMP - 1) &
```

Comment:
Looks good @kklobe!

With this commit in place, try giving `./script/format-commit.sh` a run - it will apply a couple formatting adjustments (just to this commit) to make it match Staging's overall formatting guidelines. 

It'll probably catch a couple things, which you can `ammend` (if they're not objectionable).


If they're not too cr

Reply:
Gives me
```
git version 2.30.1 (Apple Git-130)
clang-format version 12.0.0
Using paths relative to: /Users/kklobe/src/dosbox-staging
xargs: illegal option -- -
usage: xargs [-0opt] [-E eofstr] [-I replstr [-R replacements] [-S replsize]]
             [-J replstr] [-L number] [-n number [-x]] [-P maxprocs]
             [-s size] [utility [argument ...]]
./scripts/format-commit.sh: line 166: @@: syntax error: operand expected (error token is "@@")
```

I'll dig in and see what's going on.

Reply:
I see I missed the `(BitU)` cast, would this be the correct conversion?

```
			cache_code = reinterpret_cast<uint8_t*>((reinterpret_cast<uintptr_t>(cache_code_start_ptr) +
			                        PAGESIZE_TEMP - 1) &
			                       ~(PAGESIZE_TEMP - 1));
```

That looks very clunky.

Reply:
Yes - looks right.  I did some tinkering with different rounding techniques and with an intermediate `page_boundary = PAGESIZE_TEMP - 1`, but it didn't really improve it much given it's still cast-heavy.

The formatter might improve the layout a bit to make it more readable.

Reply:
Came across this C++11 function: https://en.cppreference.com/w/cpp/memory/align

I think it just performs the calculations and adjustment without any actual allocation.. so it might be a good fit.  I like that it even gives you the amount of bytes 'lost' by aligning upward to the next boundary. 

I will have to do some side by side comparisons and see if we can get the same numbers from it.

Reply:
I think the `mmap` call returns an aligned address, and for the fallthrough malloc case, maybe https://en.cppreference.com/w/c/memory/aligned_alloc would work, then the explicit re-alignment could be avoided altogether?

Reply:
Indeed; the reinterpret casts are good.. and no added reason to deviate from upstream here.